tests/phpunit/integration/Core/Modules/ModulesTest.php

```diff
@@ -62,6 +62,44 @@ function ( $instance ) {
 		);
 	}
 
+	public function test_get_available_modules__missing_dependency() {
+		$modules = new Modules( new Context( GOOGLESITEKIT_PLUGIN_MAIN_FILE ) );
+
+		add_filter(
+			'googlesitekit_available_modules',
+			function( $modules ) {
+				return array_filter(
+					$modules,
+					function( $module ) {
+						// Remove Analytics from the list of available modules.
+						return 'analytics' !== $module;
+					}
+				);
+			}
+		);
+
+		$available = array_map(
+			function ( $instance ) {
+				return get_class( $instance );
+			},
+			$modules->get_available_modules()
+		);
+
+		// Analytics is no longer present due to the filter above.
+		// Optimize is no longer present due to its dependency on Analytics.
+		$this->assertEqualSets(
```

Comment:
This should use `assertSameSetsWithIndex` as it's an associative array. `assertEqualSets` treats them as lists and keys won't be considered in the comparison. That probably doesn't really make a difference here but the intention seems to be that it would, otherwise I would expect the assertion to use a list of values instead of keyed values.

Reply:
Thanks for pointing that out. I had copied this from `test_get_available_modules`, so I have updated the usage there and in a number of other applicable places within this file.

Reply:
Further on this - having updated to use `assertSameSetsWithIndex` I found the test suite failing on the [PHP (PHP 5.6, WordPress 4.7)](https://github.com/google/site-kit-wp/runs/8022066714?check_suite_focus=true#logs) workflow with an error:

> PHP Fatal error:  Call to undefined method Google\Site_Kit\Tests\Core\Modules\ModulesTest::assertSameSetsWithIndex() in /home/runner/work/site-kit-wp/site-kit-wp/tests/phpunit/integration/Core/Modules/ModulesTest.php on line 50

I have noticed that `assertSameSetsWithIndex` is not currently in use anywhere else in the codebase - I wonder if this is why?

I'll update to `assertEqualSetsWithIndex` which I can see in common use, but interested if you have any thoughts on this one.

Reply:
Oops, that's probably what I meant. Looks like the `same` variant was only added in `5.6.0` where the `equals` ones have been around much longer. Thanks!